mlrun/serving/routers.py

```diff
@@ -195,6 +258,8 @@ def _resolve_route(self, body, urlpath):
 
     def _handle_event(self, event):
         name, route, subpath = self._resolve_route(event.body, event.path)
+        self._update_background_task_state(event)
```

Comment:
I think all the the router uses `_pre_handle_event` so lets put `_update_background_task_state` only in there.. what do you think?